src/cljs/org/broadinstitute/firecloud_ui/common/modal.cljs

```diff
@@ -27,31 +36,83 @@
-                          [:div {:style {:backgroundColor "white"}}
-                           child]]]])
-                     stack))])
+        (map (fn [child]
+               [:div {}
+                ; CodeMirror uses zIndex values up to 5; make sure that modals appear on top
+                [:div {:style {:position "fixed" :zIndex 6
+                               :top 0 :bottom 0 :left 0 :right 0
+                               :backgroundColor "rgba(110,110,110,0.4)"}}]
+                [:div {:style {:position "absolute" :zIndex 6
+                               :top (+ (aget js/document "body" "scrollTop") 30)
+                               :left 0
+                               :width "100%"}}
+                 [:div {:style {:display "flex" :justifyContent "center"}}
+                  [:div {:style {:backgroundColor "white"}}
+                   child]]]])
+             stack))])
    :component-did-mount
    (fn [{:keys [locals]}]
      (swap! locals assoc :keydown-handler (common/create-key-handler [:esc] pop-modal))
      (.addEventListener js/window "keydown" (:keydown-handler @locals)))
    :component-will-unmount
    (fn [{:keys [locals]}]
      (.removeEventListener js/window "keydown" (:keydown-handler @locals)))})
+
+
+(react/defc OKCancelForm
+  {:get-default-props
+   (fn []
+     {:show-cancel? true})
+   :render
+   (fn [{:keys [props]}]
+     (let [{:keys [header content ok-button show-cancel? cancel-text]} props
+           cancel-text (or cancel-text "Cancel")]
+       [:div {}
+        [:div {:style {:borderBottom style/standard-line
+                       :padding "20px 48px 18px"
+                       :fontSize "137%" :fontWeight 400 :lineHeight 1}}
+         header]
+        [:div {:style {:padding "22px 48px 40px" :backgroundColor (:background-gray style/colors)}}
+         content
+         (when (or show-cancel? ok-button)
+           [:div {:style {:marginTop 40 :textAlign "center"}}
+            (when show-cancel?
+              [:a {:className "cancel"
+                   :style {:marginRight (when ok-button 27) :marginTop 2 :padding "0.5em"
+                           :display "inline-block"
+                           :fontSize "106%" :fontWeight 500 :textDecoration "none"
+                           :color (:button-blue style/colors)}
+                   :href "javascript:;"
+                   :onClick pop-modal
+                   :onKeyDown (common/create-key-handler [:space :enter] pop-modal)}
+               cancel-text])
+            (cond (fn? ok-button) [comps/Button {:text "OK" :onClick ok-button}]
+                  (map? ok-button) [comps/Button ok-button]
+                  :else ok-button)])]]))
+   :component-did-mount
+   (fn [{:keys [props]}]
+     (when-let [get-first (:get-first-element-dom-node props)]
+       (common/focus-and-select (get-first))
+       (when-let [get-last (:get-last-element-dom-node props)]
```

Comment:
should :get-last-element-dom-node default to the ok button? might save some boilerplate in calling code.